server/src/main/java/de/uftos/services/ServerService.java

```diff
@@ -77,6 +93,43 @@ public void setTimetableMetadata(TimetableMetadata timetableMetadata) {
     Server server = this.repository.findAll().getFirst();
     server.setTimetableMetadata(timetableMetadata);
 
+    this.updateTimeslots(timetableMetadata);
+
     this.repository.save(server);
   }
+
+  private void updateTimeslots(TimetableMetadata timetableMetadata) {
+    List<Timeslot> timeslots = new LinkedList<>();
+
+    for (Weekday weekday : Weekday.values()) {
+      for (int i = 0; i < timetableMetadata.getTimeslotsAmount(); i++) {
+        timeslots.add(new Timeslot(weekday, i, List.of()));
+      }
+    }
+
+    List<Timeslot> currentSlots = this.timeslotRepository.findAll();
+
+    if (timeslots.size() == currentSlots.size()) {
+      return;
+    }
+
+    if (timeslots.size() > currentSlots.size()) {
+      timeslots.removeIf(slot -> slot.getSlot() < currentSlots.size() / Weekday.values().length);
+
+      this.timeslotRepository.saveAll(timeslots);
+      return;
+    }
+
+    currentSlots.removeIf(slot -> slot.getSlot() < timeslots.size() / Weekday.values().length);
+
+    List<Timetable> timetables = currentSlots.stream()
+        .flatMap(timeslot -> timeslot.getLessons().stream().map(Lesson::getTimetable)).distinct()
+        .toList();
+    this.timeslotRepository.deleteAll(currentSlots);
+    this.timetableRepository.deleteAll(timetables);
+
+    // TODO delete instances that used timeslots that don't exist anymore
+
+    this.timeslotRepository.saveAll(timeslots);
```

Comment:
What is this line for?
New timeslots are already saved in line 119.
Saving here would create new timeslots for already exisiting timeslots as the created timeslots don`t have ids.
And as there are no fields of the timeslots which require updating I think this line should simply be removed.

Reply:
Oh yeah missed that, good catch!